### ISSUE TITLE
DELETE method not working with nitro proxy

### ISSUE THREAD

**Issue by michaelcoll** [nuxt/nuxt#23422]
### Environment

- Operating System: `Linux`
- Node Version:     `v20.5.1`
- Nuxt Version:     `3.7.3`
- CLI Version:      `3.9.0`
- Nitro Version:    `2.6.3`
- Package Manager:  `pnpm@8.8.0`
- Builder:          `-`
- User Config:      `devtools`, `nitro`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

- Clone repo : https://github.com/michaelcoll/nuxt-proxy-error
- Launch app : `pnpm run dev`
- Execute command : 
```shell
curl --location --request DELETE 'localhost:3000/api/v1/delete/2'
```

### Describe the bug

Using a dummy online api, I've configured an empty nuxt app that only have a proxy configured.

The proxy work well with method : GET, POST, PUT, HEAD, OPTIONS and PATCH

But DELETE generates an error.

### Additional context

Calling directly the proxied url works (at least now when I write this issue).
```shell
curl --location --request DELETE 'https://dummy.restapiexample.com/api/v1/delete/2'
```

### Logs

```shell-script
[nuxt] [request error] [unhandled] [500] fetch failed
  at Object.fetch (node:internal/deps/undici/undici:11576:11)  
  at async sendProxy (./node_modules/.pnpm/h3@1.8.2/node_modules/h3/dist/index.mjs:1034:20)  
  at async Object.handler (./node_modules/.pnpm/h3@1.8.2/node_modules/h3/dist/index.mjs:1630:19)  
  at async Server.toNodeHandle (./node_modules/.pnpm/h3@1.8.2/node_modules/h3/dist/index.mjs:1840:7)
```

**Comment by michaelcoll** [comment]
The error is caused by this issue : https://github.com/nodejs/undici/issues/2046

**Comment by danielroe** [comment]
Would you raise this upstream on https://github.com/unjs/nitro or https://github.com/unjs/httpxy ?

**Comment by pi0** [comment]
Possibly related: https://github.com/unjs/httpxy/issues/7

Feel free to also open an issue in nitro with reproduction and thanks for reporting 👍🏼

**Comment by januarfonti** [comment]
I downgraded the node version to 16 from 18 and the problem was resolved, but I was wondering if there was another solution besides downgrading the node version?

**Comment by jasenmichael** [comment]
https://github.com/unjs/h3/pull/605

**Comment by jasenmichael** [comment]
> ### Logs
> 
> ```shell
> [nuxt] [request error] [unhandled] [500] fetch failed
>   at Object.fetch (node:internal/deps/undici/undici:11576:11)  
>   at async sendProxy (./node_modules/.pnpm/h3@1.8.2/node_modules/h3/dist/index.mjs:1034:20)  
>   at async Object.handler (./node_modules/.pnpm/h3@1.8.2/node_modules/h3/dist/index.mjs:1630:19)  
>   at async Server.toNodeHandle (./node_modules/.pnpm/h3@1.8.2/node_modules/h3/dist/index.mjs:1840:7)
> ```

@michaelcoll  I made a [pr](https://github.com/unjs/h3/pull/605) to fix this ☝️, in the meantime you can patch `h3/dist/index.mjs` by adding:

``` js
  // PATCH: force DELETE to have empty object and content-length header
  if (method === "DELETE" && body === void 0) {
    body = Buffer.from("{}");
    fetchHeaders["content-length"] = 2;
  }
  // PATCH: force DELETE to have empty object and content-length header
```

just above line 1067 (at the time of this writing):
``` js
return sendProxy(event, target, {
```

**Comment by ahoiroman** [comment]
I can confirm that the PR fixes the issue using node v20.10.0
